overlays/bootstrap.nix

```diff
@@ -313,7 +315,30 @@ in {
         })));
 
     ghc = final.haskell-nix.compiler.ghc865;
-    inherit (final.buildPackages.haskell-nix.bootstrap.packages) cabal-install alex happy;
+    cabal-install =
+      # Make sure that `cabal-install` used building `cabal-install`
+      # always has `checkMaterialization = false` to avoid infinite
+      # recursion.
+      let
+        f = check: (final.buildPackages.haskell-nix.tool "cabal" ({
+          version = "3.2.0.0";
+          index-state = final.haskell-nix.internalHackageIndexState;
+          cabal-install = f false;
+          materialized = ../materialized/cabal-install;
+        } // final.lib.optionalAttrs (!check) {
+          checkMaterialization = false;
+        })) // { version = "3.2.0.0"; };
```

Comment:
duplicated?

Reply:
I tacked the missing `version` number on to the component as #615 is still building on hydra.  It was going to be needed to work around the `assert` in `cabal-cabal-project-to-nix.nix` that I had to remove in the end anyway (because it was too strict and was another source of infinite recursion during `checkMaterialization = true`).  So we can safely remove it now anyway.